src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,155 +6,271 @@
+      const { title, description } = serializedState.rawState;
+
+      if (linksSerializeStateIsByReference(state)) {
+        const attributes = await deserializeLinksSavedObject(state);
+        return {
+          ...attributes,
+          title,
+          description,
+        };
+      }
+
+      const { attributes: attributesWithInjectedIds } = injectReferences({
+        attributes: state.attributes,
+        references: serializedState.references ?? [],
+      });
+
+      const resolvedLinks = await resolveLinks(attributesWithInjectedIds.links ?? []);
+
+      return {
+        title,
+        description,
+        links: resolvedLinks,
+        layout: attributesWithInjectedIds.layout,
+        defaultPanelTitle: attributesWithInjectedIds.title,
+        defaultPanelDescription: attributesWithInjectedIds.description,
+      };
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const error$ = new BehaviorSubject<Error | undefined>(state.error);
+      if (!isParentApiCompatible(parentApi)) error$.next(new PanelIncompatibleError());
+
+      const links$ = new BehaviorSubject<ResolvedLink[] | undefined>(state.links);
+      const layout$ = new BehaviorSubject<LinksLayoutType | undefined>(state.layout);
+      const defaultPanelTitle = new BehaviorSubject<string | undefined>(state.defaultPanelTitle);
+      const defaultPanelDescription = new BehaviorSubject<string | undefined>(
+        state.defaultPanelDescription
+      );
+      const savedObjectId$ = new BehaviorSubject(state.savedObjectId);
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          blockingError: error$,
+          defaultPanelTitle,
+          defaultPanelDescription,
+          isEditingEnabled: () => Boolean(error$.value === undefined),
+          libraryId$: savedObjectId$,
+          getTypeDisplayName: () => DISPLAY_NAME,
+          getByValueRuntimeSnapshot: () => {
+            const snapshot = api.snapshotRuntimeState();
+            delete snapshot.savedObjectId;
+            return snapshot;
+          },
+          serializeState: async (): Promise<SerializedPanelState<LinksSerializedState>> => {
+            if (savedObjectId$.value !== undefined) {
+              const linksByReferenceState: LinksByReferenceSerializedState = {
+                savedObjectId: savedObjectId$.value,
+                ...serializeTitles(),
+              };
+              return { rawState: linksByReferenceState, references: [] };
+            }
+            const runtimeState = api.snapshotRuntimeState();
+            const { attributes, references } = serializeLinksAttributes(runtimeState);
+            const linksByValueState: LinksByValueSerializedState = {
+              attributes,
+              ...serializeTitles(),
+            };
+            return { rawState: linksByValueState, references };
```

Comment:
Not entirely sure what is happening here - but, when creating a by-value links panel from scratch and saving the dashboard, the link's `label` is overwritten with the dashboard title? Which means the titles will become out-of-sync


https://github.com/elastic/kibana/assets/8698078/1272557c-d6fd-42a6-a430-b3e32290c62e

Editing works as expected 🤷

Reply:
Fixed and added tests in 9efb1a6f709e7cb6c75ee291dd1764aac02fadae